### PR TITLE
fix a bug when rendering subcharts

### PR DESCRIPTION
**Pull request by williepaul**
- in some cases, subcharts could not be rendered properly
- change how we render values such that default capabilities are set, preventing rendering errors
- set lintmode to true to allow ignoring of certain rendering errors from within the rendering engine
- add resource count checking to load-dir unit tests
- update error messages to make sure we are including the error data

### CODE REVIEW

File: pkg/iac-providers/helm/v3/load-dir.go [hunk 1]
```diff
@@ -124,7 +123,7 @@ func (h *HelmV3) createHelmChartResource(chartPath string, chartData map[string]
 
 	configData := make(map[string]interface{})
 	if err = json.Unmarshal(jsonData, &configData); err != nil {
-		logger.Warn("unable to unmarshal normalized config data")
+		logger.Warn("unable to unmarshal normalized config data", zap.Error(err))
 		logger.Debug("failed config data", zap.Any("config", configData))
```
Comment (dev-gaur) [inline pkg/iac-providers/helm/v3/load-dir.go:127]:
```suggestion
		logger.Error("failed config data", zap.Any("config", configData))
```

File: pkg/iac-providers/helm/v3/load-dir.go [hunk 2]
```diff
@@ -248,29 +251,29 @@ func (h *HelmV3) loadChart(chartPath string) ([]*utils.IacDocument, helmChartDat
 	valuesFile := filepath.Join(chartDir, helmValuesFilename)
 	fileInfo, err = os.Stat(valuesFile)
 	if err != nil {
-		logger.Warn("unable to stat values.yaml")
+		logger.Warn("unable to stat values.yaml", zap.Error(err))
```
Comment (dev-gaur) [inline pkg/iac-providers/helm/v3/load-dir.go:254]:
```suggestion
		logger.Error("unable to stat values.yaml", zap.Error(err))
```

File: pkg/iac-providers/helm/v3/load-dir.go [hunk 3]
```diff
@@ -248,29 +251,29 @@ func (h *HelmV3) loadChart(chartPath string) ([]*utils.IacDocument, helmChartDat
...
+		logger.Error("unable to unmarshal values.yaml", zap.Error(err))
 		return iacDocuments, chartMap, err
 	}
 
 	// for each template file found, render and save an iacDocument
 	var templateFileMap map[string][]*string
 	templateFileMap, err = utils.FindFilesBySuffix(filepath.Join(chartDir, helmTemplateDir), h.getHelmTemplateExtensions())
 	if err != nil {
-		logger.Warn("error while calling FindFilesBySuffix")
+		logger.Warn("error while calling FindFilesBySuffix", zap.Error(err))
```
Comment (dev-gaur) [inline pkg/iac-providers/helm/v3/load-dir.go:276]:
```suggestion
		logger.Error("error while calling FindFilesBySuffix", zap.Error(err))
```

File: pkg/iac-providers/helm/v3/testdata/happy-path-with-subchart/a-sub-chart/.helmignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+*.swp
+*.bak
+*.tmp
+*.orig
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
+.vscode/
```
Comment (dev-gaur) [inline pkg/iac-providers/helm/v3/testdata/happy-path-with-subchart/a-sub-chart/.helmignore:23]: Can we also add vendor/ here?

**Review by dev-gaur — CHANGES_REQUESTED** [review]
Minor changes requested. Also can you please change the Title ? I think its more than a bug fix ;)

File: pkg/iac-providers/helm/v3/load-dir.go [hunk 5]
```diff
@@ -81,7 +80,7 @@ func (h *HelmV3) LoadIacDir(absRootDir string) (output.AllResourceConfigs, error
 		var config *output.ResourceConfig
 		config, err = h.createHelmChartResource(chartPath, chartMap)
 		if err != nil {
-			logger.Debug("failed to create helm chart resource", zap.Any("config", config))
+			logger.Debug("failed to create helm chart resource", zap.Any("config", config), zap.Error(err))
```
Comment (kanchwala-yusuf) [inline pkg/iac-providers/helm/v3/load-dir.go:83]: Can we make this an `Error` level log?

Comment (kanchwala-yusuf) [reply pkg/iac-providers/helm/v3/load-dir.go:83]: Or may be `Warn`?

File: pkg/iac-providers/helm/v3/load-dir.go [hunk 6]
```diff
@@ -192,23 +191,27 @@ func (h *HelmV3) renderChart(chartPath string, chartMap helmChartData, templateD
...
 	var renderData map[string]string
-	renderData, err = engine.Render(c, v)
+	var e engine.Engine
+
+	// lint mode tells the rendering engine to be less strict when it comes to required variables
+	e.LintMode = true
+	renderData, err = e.Render(c, v)
 	if err != nil {
-		logger.Warn("error encountered while rendering chart", zap.String("template dir", templateDir))
+		logger.Warn("error encountered while rendering chart", zap.String("template dir", templateDir), zap.Error(err))
```
Comment (kanchwala-yusuf) [inline pkg/iac-providers/helm/v3/load-dir.go:214]: I believe, this can be an `Error` level log?

**Review by kanchwala-yusuf — CHANGES_REQUESTED** [review]

File: pkg/iac-providers/helm/v3/load-dir.go (hunk 5 above)
Comment (williepaul) [reply pkg/iac-providers/helm/v3/load-dir.go:83]: right--should be error.

File: pkg/iac-providers/helm/v3/load-dir.go (hunk 1 above)
Comment (williepaul) [reply pkg/iac-providers/helm/v3/load-dir.go:127]: done

File: pkg/iac-providers/helm/v3/load-dir.go (hunk 6 above)
Comment (williepaul) [reply pkg/iac-providers/helm/v3/load-dir.go:214]: done

File: pkg/iac-providers/helm/v3/load-dir.go (hunk 2 above)
Comment (williepaul) [reply pkg/iac-providers/helm/v3/load-dir.go:254]: done

**Review by kanchwala-yusuf — APPROVED** [review]

**Pull request merged**